eng/pipelines/templates/steps/cmake-build.yml

```diff
@@ -13,6 +13,6 @@ steps:
     workingDirectory: build
     displayName: cmake generate
 
-  - script: cmake --build .
+  - script: cmake --build . --config Release
```

Comment:
@danieljurek - what do we have to do to have some legs run just `cmake --build .` and some that run `cmake --build . --config Release`.

I don't want all the legs to run the release configuration.

There are four permutations here:
- --config debug, CMAKE_BUILD_TYPE=Debug
- --config release, CMAKE_BUILD_TYPE=Debug
- --config debug, CMAKE_BUILD_TYPE=Release
- --config release, CMAKE_BUILD_TYPE=Release

Reply:
I recommend that we use a parameter in `cmake-build.yml` because the value will be the same (docs don't say it but from anecdotal experience last week `--config` values can use the same casing). 

```yaml
parameters:
  ...
  Configuration: Debug

steps:
  ...
  - script: cmake .. ${{ parameters.BuildArgs }} -DCMAKE_BUILD_TYPE=${{ parameters.Configuration }} 
  ...
  - script: cmake --build . --config ${{ parameters.Configuration }} 
  ...
```

When you invoke the template: 

```yaml
...
jobs: 
  variables: 
    # Use Debug by default unless another build type is specified
    BuildConfiguration: Debug
  ...
  strategy:
    matrix:
      MatrixEntry1: 
        ...
        BuildConfiguration: Release
        ...

  steps:
    ...
    - template: cmake-build.yml
      parameters: 
        Configuration: $(BuildConfiguration) 
        ...
```

Reply:
@ahsonkhan can you open a bug to track this work (if there isn't already one).  Mark it as september and high priority.  We need our pipelines rock solid before GA

Reply:
> can you open a bug to track this work (if there isn't already one).

https://github.com/Azure/azure-sdk-for-c/issues/1113